src/category_theory/monoidal/category.lean

```diff
@@ -0,0 +1,405 @@
+    (associator (tensor_unit C) X Y).hom :=
+begin
+  conv_rhs { rw [associator_naturality] }
+end
+
+lemma left_unitor_product_aux (X Y : C) :
+    ((𝟙 (tensor_unit C)) ⊗ (associator (tensor_unit C) X Y).hom) ≫
+    ((𝟙 (tensor_unit C)) ⊗ (left_unitor (X ⊗ Y)).hom)
+  = (𝟙 (tensor_unit C)) ⊗ ((left_unitor X).hom ⊗ (𝟙 Y)) :=
+begin
+  rw ←(cancel_epi (associator (tensor_unit C) ((tensor_unit C) ⊗ X) Y).hom),
+  rw left_unitor_product_aux_square,
+  rw ←(cancel_epi ((associator (tensor_unit C) (tensor_unit C) X).hom ⊗ (𝟙 Y))),
+  slice_rhs 1 2 { rw left_unitor_product_aux_triangle },
+  conv_lhs { rw [left_unitor_product_aux_perimeter] }
+end
+
+lemma right_unitor_product_aux_perimeter (X Y : C) :
+    ((associator X Y (tensor_unit C)).hom ⊗ (𝟙 (tensor_unit C))) ≫
+    (associator X (Y ⊗ (tensor_unit C)) (tensor_unit C)).hom ≫
+    ((𝟙 X) ⊗ (associator Y (tensor_unit C) (tensor_unit C)).hom) ≫
+    ((𝟙 X) ⊗ (𝟙 Y) ⊗ (left_unitor (tensor_unit C)).hom)
+  = ((right_unitor (X ⊗ Y)).hom ⊗ (𝟙 (tensor_unit C))) ≫
+    (associator X Y (tensor_unit C)).hom :=
+begin
+  transitivity (((associator X Y _).hom ⊗ 𝟙 _) ≫ (associator X _ _).hom ≫
+    (𝟙 X ⊗ (associator Y _ _).hom)) ≫
+    (𝟙 X ⊗ 𝟙 Y ⊗ (monoidal_category.left_unitor _).hom),
+  { conv_lhs { congr, skip, rw [←category.assoc] },
+    conv_rhs { rw [category.assoc] } },
+  { conv_lhs { congr, rw [monoidal_category.pentagon] },
+    conv_rhs { congr, rw [←monoidal_category.triangle] },
+    conv_rhs { rw [category.assoc] },
+    conv_rhs { congr, skip, congr, congr, rw [←tensor_map_id] },
+    conv_rhs { congr, skip, rw [associator_naturality] },
+    conv_rhs { rw [←category.assoc] } }
+end
+
+lemma right_unitor_product_aux_triangle (X Y : C) :
+    ((𝟙 X) ⊗ (associator Y (tensor_unit C) (tensor_unit C)).hom) ≫
+    ((𝟙 X) ⊗ (𝟙 Y) ⊗ (left_unitor (tensor_unit C)).hom)
+  = (𝟙 X) ⊗ (right_unitor Y).hom ⊗ (𝟙 (tensor_unit C)) :=
+begin
+  conv_lhs { rw [←interchange_right_identity] },
+  conv_rhs { congr, skip, rw [←monoidal_category.triangle] }
+end
+
+lemma right_unitor_product_aux_square (X Y : C) :
+    (associator X (Y ⊗ (tensor_unit C)) (tensor_unit C)).hom ≫
+    ((𝟙 X) ⊗ (right_unitor Y).hom ⊗ (𝟙 (tensor_unit C)))
+  = (((𝟙 X) ⊗ (right_unitor Y).hom) ⊗ (𝟙 (tensor_unit C))) ≫
+    (associator X Y (tensor_unit C)).hom :=
+begin
+  conv_rhs { rw [associator_naturality] }
+end
+
+lemma right_unitor_product_aux (X Y : C) :
+    ((associator X Y (tensor_unit C)).hom ⊗ (𝟙 (tensor_unit C))) ≫
+    (((𝟙 X) ⊗ (right_unitor Y).hom) ⊗ (𝟙 (tensor_unit C)))
+  = ((right_unitor (X ⊗ Y)).hom ⊗ (𝟙 (tensor_unit C))) :=
+begin
+  rw ←(cancel_mono (associator X Y (tensor_unit C)).hom),
+  slice_lhs 2 3 { rw ←right_unitor_product_aux_square },
+  conv_lhs { congr, skip, congr, skip, rw [←right_unitor_product_aux_triangle] },
+  conv_rhs { rw [←right_unitor_product_aux_perimeter] }
+end
+
+-- See Proposition 2.2.4 of http://www-math.mit.edu/~etingof/egnobookfinal.pdf
+@[simp] lemma left_unitor_product (X Y : C) :
```

Comment:
Should this maybe be called `left_unitor_tensor`?

Reply:
Agreed. Fixed.